packages/waku/src/router/create-pages.ts

```diff
@@ -753,7 +753,38 @@ export const createPages = <
           };
         },
       );
-      return [...routeConfigs, ...apiConfigs];
+      return (
+        [...routeConfigs, ...apiConfigs]
+          // Sort routes by priority: "standard routes" -> api routes -> api wildcard routes -> standard wildcard routes
+          .sort((pathConfigA, pathConfigB) => {
+            const hasWildcardA = pathConfigA.path.at(-1)?.type === 'wildcard';
+            const hasWildcardB = pathConfigB.path.at(-1)?.type === 'wildcard';
+
+            let priorityA = 0;
+            if (pathConfigA.type === 'api') {
+              if (hasWildcardA) {
+                priorityA = 2;
+              } else {
+                priorityA = 1;
+              }
+            } else if (hasWildcardA) {
+              priorityA = 3;
+            }
```

Comment:
Make it a function, incl. L760, and reuse the logic. It will be more readable.

Reply:
This commit abstracts this logic away into a function: https://github.com/wakujs/waku/pull/1451/commits/a4a6e63a1a754f481d68eccd4026767d91d14f19